core/state/state_object.go

```diff
@@ -193,6 +191,7 @@ func (s *stateObject) GetCommittedState(key common.Hash) common.Hash {
 	//      have been handles via pendingStorage above.
 	//   2) we don't have new values, and can deliver empty response back
 	if _, destructed := s.db.stateObjectsDestruct[s.address]; destructed {
+		s.originStorage[key] = common.Hash{} // track the empty slot as origin value
```

Comment:
Any particular reason for starting to track this here?

Reply:
Yeah, i think it would be better to track the "empty" value as origin, reasons:

Here, https://github.com/rjl493456442/go-ethereum/blob/state-update/core/state/state_object.go#L428 we blindly assume the original value is tracked in the `s.originStorage` map. It would still be correct if we don't track empty value, but it feels pretty weird.

Besides, it aligns with the behavior in https://github.com/rjl493456442/go-ethereum/blob/state-update/core/state/state_object.go#L233, as the slot read from database could also be empty.

What's more, tracking the empty value here could potentially be useful for differentiating "existence" and "emptiness" in the future? e.g. we could error out if a slot is never accessed (not existent in the originStorage map) but later be modified?